src/controller/submissions/submission.service.ts

```diff
@@ -875,4 +878,168 @@ export class SubmissionService {
+        limit: limit,
+        offset: offset
+      });
+      const totalStudents = await db.select().from(zuvyChapterTracking).where(sql`${zuvyChapterTracking.chapterId} = ${chapterId}`);
+      const totalStudentsCount = totalStudents.length;
+      const totalPages = Math.ceil(totalStudentsCount / limit);
+      const deadlineDate = new Date(chapterDeadline[0].deadline).getTime();
+
+      const data = statusOfStudentCode.map((statusCode) => {
+      const studentAssignmentStatus = statusCode;
+      let isLate = false;
+
+     if (studentAssignmentStatus && studentAssignmentStatus['completedAt']) {
+         const createdAtDate = new Date(studentAssignmentStatus['completedAt']).getTime();
+     if (createdAtDate > deadlineDate) {
+        isLate = true;
+       }
+     }
+
+      return {
+        id: Number(statusCode['user']['id']),
+        name: statusCode['user']['name'],
+        emailId: statusCode['user']['email'],
+         status: isLate ? 'Late Submission' : 'On Time',
+        };
+     });
+
+     const currentPage =!isNaN(limit) && !isNaN(offset) ? offset/limit + 1 : 1;
+
+      return {status: helperVariable.success,code: STATUS_CODES.OK, data, totalPages, totalStudentsCount,currentPage };
+    } catch (error) {
+      return [new ErrorResponse(error.message, STATUS_CODES.BAD_REQUEST, false)]
+    }
+  }
+   
+
+   async getAssignmentSubmissionDetailForUser(chapterId:number,userId:number)
+   {
+    try{
+      const assignmentDetails = await db.query.zuvyModuleChapter.findFirst({
+        where: (moduleChapter, { eq }) =>
+          eq(moduleChapter.id, chapterId),
+        columns: {
+          id:true,
+          topicId:true,
+          articleContent:true,
+          completionDate:true
+        },
+        with: {
+          chapterTrackingDetails: {
+            columns: {
+              completedAt:true
+            },
+            where : (chapterTracking,{eq}) =>
+              eq(chapterTracking.userId,BigInt(userId)),
+            with: {
+              user: {
+                columns: {
+                  id: true,
+                  name: true,
+                  email: true,
+                },
+                with: {
+                  studentAssignmentStatus : {
+                    columns : {
+                      projectUrl:true
+                    }
+                  }
+                }
+              },
+            },
+          }
+        },
+      })
+
+      let isSubmittedOnTime = true;
+       if(new Date(assignmentDetails['completionDate']).getTime() < new Date(assignmentDetails['chapterTrackingDetails'][0]['completedAt']).getTime())
+         {
+           isSubmittedOnTime = false;
+        }
+         assignmentDetails['chapterTrackingDetails'][0]['user']['id'] = Number( assignmentDetails['chapterTrackingDetails'][0]['user']['id'] ) 
+         assignmentDetails['chapterTrackingDetails'][0]['status'] = isSubmittedOnTime == true ? 'Submitted on time' : 'Submitted late';
+        return {
```

Comment:
Inside the /errorHandler/handler, could you create the success response object? If we’re returning a success message, we should call that success response function.